Source/Core/InputCommon/ControllerInterface/evdev/evdev.cpp

```diff
@@ -371,6 +378,21 @@ void Init()
 
 void PopulateDevices()
 {
+  // Unfortunately there doesn't seem to be a documented way to
+  // enumerate devices in a deterministic order.
+  // This is especially problematic for 4-port controller adapters.
+  // Sorting device nodes based on their /dev/ name seems to work.
+  std::map<std::string, std::string> devnodes;
+
+  // Pad the first occurance of digits for natural sorting,
+  // so /dev/input/event9 sorts before /dev/input/event10.
```

Comment:
Would it make more sense to sort by /dev/input/by-id/ or /dev/input/by-path/ ? According to what I can gather /dev/input/eventX is by order of discovery.

Reply:
Order of discovery is desired. The first attached device should take the first slot and so on. A second device should never pop in before the first. I've updated the comment to make that more clear.